extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/KnownPathResourceManager.java

```diff
@@ -69,6 +66,9 @@ public Resource getResource(String path) throws IOException {
         if (directories.contains(path)) {
             return new DirectoryResource(path);
         }
+        if (!files.contains(path)) {
+            return null;
+        }
```

Comment:
The fix is actually here. The rest is to make paths consistent so that we don't end up comparing Unix-styled paths and Windows-style paths.